addons/containerd/1.3.7/install.sh

```diff
@@ -178,7 +178,7 @@ function containerd_migrate_from_docker() {
 
     local kubeconfigFlag="--kubeconfig=/etc/kubernetes/kubelet.conf"
 
-    local node=$(hostname | tr '[:upper:]' '[:lower:]')
+    local node=$(cat /proc/sys/kernel/hostname | tr '[:upper:]' '[:lower:]')
```

Comment:
See the k8s implementation and comment:

https://github.com/kubernetes/kubernetes/blob/f58f70bd5730658505042cd9baa80f72d3b6e31e/staging/src/k8s.io/component-helpers/node/util/hostname.go#L38-L40

Then, see the code source for os.Hostname() to linux platforms:

https://github.com/golang/go/blob/2ebe77a2fda1ee9ff6fd9a3e08933ad1ebaea039/src/os/sys_linux.go#L12-L53

And the line 38 from the link that you added : https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/os/sys_linux.go;l=38

So, I think we should stick with `cat /proc/sys/kernel/hostname` . Try uname first, shows that is required because it will run in other platforms such as Android.

Reply:
In this PR we are adding the command and in a follow up we will use it.